packages/plugins/flow/operations/src/index.ts

```diff
@@ -10,25 +10,38 @@ export const plugin: PluginFunction<FlowDocumentsPluginConfig> = (
   config: FlowDocumentsPluginConfig
 ) => {
   const documents = config.flattenGeneratedTypes ? optimizeOperations(schema, rawDocuments) : rawDocuments;
-  const prefix = `type $Pick<Origin: Object, Keys: Object> = $ObjMapi<Keys, <Key>(k: Key) => $ElementType<Origin, Key>>;\n`;
 
-  const allAst = concatAST(documents.map(v => v.document));
+  const prefix = config.preResolveTypes
+    ? ''
+    : `type $Pick<Origin: Object, Keys: Object> = $ObjMapi<Keys, <Key>(k: Key) => $ElementType<Origin, Key>>;\n`;
+
+  let allAst = concatAST(documents.map(v => v.document));
+  const includedFragments = concatAST(rawDocuments.map(v => v.document)).definitions.filter(
+    d => d.kind === Kind.FRAGMENT_DEFINITION
+  );
+
+  // Fragments get removed when the types are optimized by relay
+  // We still want to export the types of the fragments included in the documents
+  if (config.flattenGeneratedTypes) {
```

Comment:
Maybe we should use `includeFragments` instead?
https://github.com/ardatan/graphql-tools/blob/master/packages/relay-operation-optimizer/src/index.ts#L15

Reply:
Updated to use that approach. Good find!